ykbuild/build.rs

```diff
@@ -36,12 +36,25 @@ fn main() {
         .build_arg(nprocs);
 
     if let Ok(args) = env::var("YKB_YKLLVM_BUILD_ARGS") {
+        // Caveat: this assumes no cmake argument contains a ',' or a ':'.
         for arg in args.split(",") {
-            if !arg.starts_with("-D") {
-                panic!("YKB_YKLLVM_BUILD_ARGS must only contain -D arguments");
+            let elems = arg.split(":").collect::<Vec<_>>();
+            if elems.len() != 2 {
+                panic!("Failed to parse YKB_YKLLVM_BUILD_ARGS");
+            }
+            match elems[0] {
```

Comment:
I prefer writing this sort of thing as:

```
match arg.split(":").collect::<Vec<_>>() {
  ["define", v] => ...,
  ["build_arg", v] => ...,
  ["generator", v] => ...
  [x, _] => panic!("Unknown kind {x}"),
  x => panic!("Incorrectly formatted option {x}")
}
```

because it keeps the splitting and whatnot more together. Up to you which you prefer!

Reply:
I prefer that too! I didn't know you could match like that!

Reply:
Fixed in [f3ebb76](https://github.com/ykjit/yk/pull/678/commits/f3ebb76b8b1e5baea268d2256295636da8eb4d55)

Reply:
[87ea181](https://github.com/ykjit/yk/pull/678/commits/87ea1814ca55792b690c9d46f711f2a3583d6b22) completes the fix, although the build is still failing for me locally due to the `PATH` issue we discussed in #679 (it doesn't find `llvm-config`. I guess CI will work because it sets PATH.